### ISSUE TITLE
Error in precompiling in Julia 1.0: "syntax: try without catch or finally" 

### ISSUE THREAD

**Issue by fHachenberg** [JuliaLang/PkgDev.jl#149]
What I did:

- download Julia 1.0
- start Julia
- using Pkg
- Pkg.add("PkgDev")

```
 Resolving package versions...
 Installed PkgDev ─ v0.2.1
  Updating `~/.julia/environments/v1.0/Project.toml`
 [no changes]
  Updating `~/.julia/environments/v1.0/Manifest.toml`
 [no changes]
```

- using PkgDev

-> Error

```
[ Info: Precompiling PkgDev [149e707d-584d-56d3-88ec-740c18e106ff]
ERROR: LoadError: LoadError: syntax: try without catch or finally
Stacktrace:
 [1] include at ./boot.jl:317 [inlined]
 [2] include_relative(::Module, ::String) at ./loading.jl:1038
 [3] include at ./sysimg.jl:29 [inlined]
 [4] include(::String) at /home/fabian/.julia/packages/PkgDev/DJ4wQ/src/PkgDev.jl:3
 [5] top-level scope at none:0
 [6] include at ./boot.jl:317 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1038
 [8] include(::Module, ::String) at ./sysimg.jl:29
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:319 [inlined]
 [11] eval(::Expr) at ./client.jl:389
 [12] top-level scope at ./none:3
in expression starting at /home/fabian/.julia/packages/PkgDev/DJ4wQ/src/generate.jl:127
in expression starting at /home/fabian/.julia/packages/PkgDev/DJ4wQ/src/PkgDev.jl:13
ERROR: Failed to precompile PkgDev [149e707d-584d-56d3-88ec-740c18e106ff] to /home/fabian/.julia/compiled/v1.0/PkgDev/KD6BZ.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] macro expansion at ./logging.jl:311 [inlined]
 [5] _require(::Base.PkgId) at ./loading.jl:941
 [6] require(::Base.PkgId) at ./loading.jl:852
 [7] macro expansion at ./logging.jl:311 [inlined]
 [8] require(::Module, ::Symbol) at ./loading.jl:834
 [9] #37 at ./boot.jl:319 [inlined]
 [10] with_logstate(::getfield(Main, Symbol("##37#39")), ::Base.CoreLogging.LogState) at ./logging.jl:397
 [11] with_logger(::Function, ::Atom.Progress.JunoProgressLogger) at ./logging.jl:493
 [12] top-level scope at /home/fabian/.julia/packages/Atom/jodeb/src/repl.jl:128
```

**Comment by Moelf** [comment]
what's the newest workflow of publishing a new package?

**Comment by scls19fr** [comment]
Maybe you should have a look at https://docs.julialang.org/en/latest/stdlib/Pkg/#Creating-your-own-projects-1

https://github.com/invenia/PkgTemplates.jl could be interesting but it doesn't seem to be currently available for Julia 0.7/1.0 in METADATA
https://github.com/invenia/PkgTemplates.jl/issues/22
